### PR TITLE
JAPI-00109 File Structure info

### PR DESCRIPTION
**Pull request by johnholt**
Add parameters to request file structure information to HPCCWsDFUClient.

Add new fields and missing fields to copy function in DFUFileDetailInfo class.

Signed-off-by: johnholt <john.d.holt@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [x] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [x] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

## Testing:

File structure information retrieve from files residing on the ML Dev cluster.

### CODE REVIEW

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java [hunk 1]
```diff
@@ -120,10 +120,25 @@ public WsDfuServiceSoapProxy getSoapProxy() throws Exception
...
+    }
+    /**
+     *
+     * @param logicalname logical file name, can start with ~
+     * @param clustername optional, if specified the cluster name used in the search
+     * @param wantRecordInfo want record structure information returned
+     * @return
+     * @throws Exception
+     */
+    public DFUInfoResponse getFileInfo(String logicalname, String clustername,
```
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java]: let's rename this flag to fetchRecordInfo.
In fact, let's expose both flags: fetchBinaryRecordInfo, and fetchJSONRecordInfo.
It seems unnecessary to request the record info in 2 formats.

line 124 could change to return this.getFileInfo(logicalname,  clustername, false, false)

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java [hunk 2]
```diff
@@ -687,10 +702,23 @@ public int hashCode()
...
+    /**
+     *
+     * @param logicalname logical file for request, can start with ~
+     * @param clustername optional
+     * @param wantRecordInfo true if record structure information is to be returned
+     * @return DFUInfoResponse object containing the information
+     * @throws Exception
+     */
+    public DFUFileDetailInfo getFileDetails(String logicalname,
+        String clustername, boolean wantRecordInfo) throws Exception
```
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java]: same concept here, let's add the fetchBin/JSONRecordInfo flags.

**Comment by johnholt** [comment]
@rpastrana  this should be ready to go unless you see something else.

**Comment by rpastrana** [comment]
@Michael-Gardner please re-build 2.0.0-SNAPSHOT and deploy, thanks.

**Pull request merged**